### ISSUE TITLE
New option in no-unsafe-negation to report ! a < b 

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12163]
**What rule do you want to change?**

`no-unsafe-negation`

> This rule disallows negating the left operand of Relational Operators. 

The rule currently targets only `in` and `instanceof`.

`<`, `>`, `<=`, `>=` are also [relational operators](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-relational-operators).

`! a < b` is equivalent to `(a ? 0 : 1) < b` (I think), which is rarely the intended meaning:

```js
function min(a, b) {
    let m = a;
    if (! a < b) {
        m = b;
    }
    return m;
}

min(1, 2); // 2
```

**Does this change cause the rule to produce more or fewer warnings?**

More if the option is set to `true`. Default is `false`.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New option.

**Please provide some example code that this change will affect:**

```js
/*eslint no-unsafe-negation: "error"*/

if (! a < b) {}

while (! a > b) {}

foo = ! a <= b;

foo = ! a >= b;
```

**What does the rule currently do for this code?**

Nothing.

**What will the rule do after it's changed?**

4 errors

```js
/*eslint no-unsafe-negation: ["error", { "enforceForOrderingRelations": true }]*/

if (! a < b) {} // error

while (! a > b) {} // error

foo = ! a <= b; // error

foo = ! a >= b; // error
```

**Are you willing to submit a pull request to implement this change?**

Yes.

**Comment by kaicataldo** [comment]
This has now been accepted.

**Comment by samrae7** [comment]
@mdjermanovic do you still want to do the PR for this? If not I could take it on

**Comment by mdjermanovic** [comment]
@samrae7 feel free to claim this. PR is welcome, thanks!

**Comment by samrae7** [comment]
@mdjermanovic thank you. I will work on it over the weekend
